pkg/controller/clusterdeployment/clusterinstalls.go

```diff
@@ -163,10 +166,24 @@ func (r *ReconcileClusterDeployment) reconcileExistingInstallingClusterInstall(c
 		)
 	}
 
-	completed = controllerutils.FindCondition(conditions, hivev1.ClusterInstallCompletedClusterDeploymentCondition)
-	if completed.Status == corev1.ConditionTrue { // the cluster install is complete
+	// if the cluster install stopped without being completed or failed...
+	if updated && clusterInstallFailed.Status != corev1.ConditionTrue && clusterInstallCompleted.Status != corev1.ConditionTrue {
```

Comment:
This feels like it's in a weird place. I think (and you can set up a test case to prove) we can erroneously hit this condition as follows:
- Initial state, ProvisionStopped condition is initialized/unknown
- Interim state, consumer's install is proceeding, they set ClusterInstall's Stopped condition to False/Proceeding. We'll set `updated=true` from L147; and then we'll enter this condition because their Failed and Completed conditions are still False because they're not done yet. And that'd be our bug, not theirs :)

Overall, I think this function would be more scrutable if we more linearly/tightly mapped their `Completed`/`Failed` to our `Provisioned` condition; and their `Stopped` to our `ProvisionStopped`. Like:

- If our CD's ProvisionStopped is True, we've already done whatever we're going to do. No-op and short out.
- Grab the three ClusterInstall conditions
  - If Failed && Stopped, set Provisioned=failed and ProvisionStopped=true
  - If Complete && Stopped, set Provisioned=provisioned and ProvisionStopped=true
  - If !Failed && !Complete && !Stopped, set Provisioned=provisioning (in progress) and ProvisionStopped=false
  - else error: set Provisioned=error and ProvisionStopped=true

If we're always setting the pair of conditions together, we get to keep track of exactly two "updated" bool vars and use them exclusively to decide whether to do a `Status().Update()`. We shouldn't have to use them for anything else.